src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp

```diff
@@ -85,28 +86,14 @@ struct EvolutionMetavars
                  GeneralizedHarmonic::Tags::Pi<volume_dim, frame>,
                  GeneralizedHarmonic::Tags::Phi<volume_dim, frame>>;
 
-  using observe_fields = typename GeneralizedHarmonicTemplateBase<
-      EvolutionMetavars>::observe_fields;
   struct factory_creation
       : tt::ConformsTo<Options::protocols::FactoryCreation> {
-    using factory_classes = tmpl::map<
-        tmpl::pair<
-            Event,
-            tmpl::flatten<tmpl::list<
-                Events::Completion,
-                dg::Events::field_observations<volume_dim, Tags::Time,
-                                               observe_fields,
-                                               analytic_solution_fields>,
-                Events::time_events<system>,
-                intrp::Events::Interpolate<3, AhA, interpolator_source_vars>>>>,
-        tmpl::pair<StepChooser<StepChooserUse::LtsStep>,
-                   StepChoosers::standard_step_choosers<system>>,
-        tmpl::pair<
-            StepChooser<StepChooserUse::Slab>,
-            StepChoosers::standard_slab_choosers<system, local_time_stepping>>,
-        tmpl::pair<StepController, StepControllers::standard_step_controllers>,
-        tmpl::pair<Trigger, tmpl::append<Triggers::logical_triggers,
-                                         Triggers::time_triggers>>>;
+    using factory_classes = Options::add_factory_classes<
```

Comment:
I think this could quickly accumulate levels of nested `add_factory_classes` metafunction calls when adding classes to multiple keys. Would it be reasonable to make this metafunction work a bit like `tmpl::push_back`, taking a parameter pack of key-values to add? It could even be in the `tmpl` namespace. I'm thinking something like this:

```
tmpl::push_back_at<
  original_factory_classes,
  tmpl::pair<Event, tmpl::list<...>>,
  tmpl::pair<Trigger, tmpl::list<...>>>
```

What do you think?

Reply:
I've done the generalization.  I don't think this is generic enough to move to `tmpl`, but we can reevaluate if we find another use for it.